src/models/data_table.ts

```diff
@@ -1,13 +1,19 @@
 import _, { Dictionary } from 'lodash'
-import { messages } from 'cucumber-messages'
+import { messages } from '@cucumber/messages'
 
 export default class DataTable {
   private readonly rawTable: string[][]
 
-  constructor(pickleTable: messages.PickleStepArgument.IPickleTable) {
-    this.rawTable = pickleTable.rows.map((row) =>
-      row.cells.map((cell) => cell.value)
-    )
+  constructor(
+    sourceTable: messages.PickleStepArgument.IPickleTable | string[][]
+  ) {
+    if (sourceTable instanceof Array) {
```

Comment:
What is this condition for?

Reply:
To support the `transpose` method, added via https://github.com/cucumber/cucumber-js/pull/1318/commits/7f746ae524e452cf25c60c5380354b7166520e19 - this is present in other data table impelmentations I think